src/main/scala/com/microsoft/hyperspace/util/DataAnalysisUtils.scala

```diff
@@ -0,0 +1,278 @@
+      .attr("text-anchor", "middle")
+      .attr("transform", "translate(" + (width / 2) + "," + (height - 4) + ")")
+      .style("font-size", "15px")
+      .text("Column value range");
+
+    </script>
+    <body><svg width=500 height=500></svg></body>
+    """
+  }
+
+  def minMaxAnalysis(df: DataFrame, colName: String, htmlFormat: Boolean = false): String = {
+    val dataType = df.schema.fields.find(_.name.equals(colName)).get.dataType
+    val numeric = try {
+      TypeUtils.getNumeric(dataType)
+    } catch {
+      case _: Exception =>
+        throw HyperspaceException(
+          s"minMaxAnalysis only supports Numeric type. colName=$colName, dataType=$dataType")
+    }
+
+    val distFigureLen = if (htmlFormat) 20 else 50
+    val newline = if (htmlFormat) "</br>" else System.lineSeparator()
+    val res = df.inputFiles.par.map { f =>
+      val file = df.sparkSession.read.parquet(f)
+      f -> file.agg(min(col(colName)), max(col(colName))).collect.toSeq.headOption
+    }.toIndexedSeq.filter(_._2.isDefined)
+
+    /* groupBy shuffles all data, though it can be done with one spark job.
+    val res = df
+      .groupBy(input_file_name() as "filename")
+      .agg(min(df(colName)), max(df(colName)))
+      .collect
+      .toSeq
+     */
+
+    val inputFiles = df.inputFiles
+    val conf = df.sparkSession.sessionState.newHadoopConf()
+    val sizeInBytesMap = inputFiles.par.map { path =>
+      path -> FileUtils.getDirectorySize(new Path(path), conf)
+    }.toMap
+
+    val ordering = TypeUtils.getInterpretedOrdering(dataType)
+    val range = res
+      .flatMap { row =>
+        val r = row._2.get
+        val byteSize = sizeInBytesMap(row._1) // fileName
+        (r.get(0), byteSize, r.get(1), true) :: (r.get(1), byteSize, r.get(0), false) :: Nil
+      }
+      .sortWith((a, b) => if (ordering.equals(a._1, b._1)) a._4 else ordering.lt(a._1, b._1))
+
+    val minVal = range.head._1
+    val maxVal = range.last._1
+    val unit = numeric.toDouble(numeric.minus(maxVal, minVal)) / distFigureLen.toDouble
+
+    val stringBuilder = new StringBuilder
+
+    /* File distribution
+    stringBuilder.append("+")
+    stringBuilder.append("-" * (distFigureLen))
+    stringBuilder.append("+")
+    stringBuilder.append(newline)
+    res.foreach { r =>
+      val startRange = ((r.getLong(1) - minVal) / unit).toInt
+      stringBuilder.append("|")
+      val endRange = ((r.getLong(2) - minVal) / unit).toInt + 1
+      val endRangeFixed = endRange.min(distFigureLen)
+      val startRangeFixed = startRange.min(distFigureLen - 1)
+      stringBuilder.append(" " * startRangeFixed)
+      stringBuilder.append("*" * (endRangeFixed - startRangeFixed))
+      stringBuilder.append(" " * (distFigureLen - endRangeFixed))
+      stringBuilder.append("|")
+      stringBuilder.append(newline)
+    }
+    stringBuilder.append("+")
+    stringBuilder.append("-" * (distFigureLen))
+    stringBuilder.append("+")
+    stringBuilder.append(newline)
+     */
```

Comment:
I thought this PR is ready to be merged once reviewed? Should this comment be committed?